### PR TITLE
[Bug] Fix for Octolock bypasses Ghost Invulnerability to lower Stats

### LINKED ISSUE #4867: [Bug] Octolock bypasses Ghost Invulnerability to lower Stats
### Describe the bug

Octolock applies the Defense and Special Defense dropping effect on Ghost-Type pokemon despite them being invulnerable to trapping.

-The Pokémon can still switch freely, however the Octolock effect does not lapse if the wave is advanced
### Reproduction

Use Octolock on a Ghost Type

### Expected behavior

https://replay.pokemonshowdown.com/gen9customgame-2238349469-7jqq8qz8c19qtfq0kghbxxbalz6pnfspw

Octolock should not work on Ghost types

### Screenshots / Videos

Octolock activates against Ghost type Pokemon, but can still switch out.

https://github.com/user-attachments/assets/9d5b37d6-f87d-4272-a6a7-0369b4d28064

Octolock does not activate against semi-invulnerable Pokemon

https://github.com/user-attachments/assets/75f2aa7d-7bb7-4537-9e01-2d1e795ea7bf

Syrup Bomb does not activate against Sap Sipper

https://github.com/user-attachments/assets/6ba56a69-4968-49b9-82f5-d7cf4bd84e41

### Session export file

_No response_

### User data export file

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PrabbyDD**
## What are the changes the user will see?
octolock properly interacts with ghost types

## Why am I making these changes?
Fixes #4867 

## What are the changes from a developer perspective?
changed some tests around and removed some obsolete ones, as well as just removing canAdd from octolock tag

## Screenshots/Videos

https://github.com/user-attachments/assets/99b5649b-82ca-42de-89ad-5fcf4804f010

https://github.com/user-attachments/assets/31c86805-d197-4171-af34-e7be3ea3fb9a

## How to test the changes?
override files, as well as tests

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?
- [x] Are all unit tests still passing? (`npm run test`)
  - [x] Have I created new automated tests (`npm run create-test`) or updated existing tests related to the PR's changes?
- [x] Have I provided screenshots/videos of the changes (if applicable)?
  - [x] Have I made sure that any UI change works for both UI themes (default and legacy)?

Are there any localization additions or changes? If so:
- [ ] Has a locales PR been created on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repo?
  - [ ] If so, please leave a link to it here: 
- [ ] Has the translation team been contacted for proofreading/translation?

### CODE REVIEW

**Review by innerthunder — CHANGES_REQUESTED** [review]
The move should fail when used against a Ghost-type instead of resolving and doing nothing.

I think the best approach to fix this would be to add a `.condition()` function to Octolock, Block, Mean Look, and Spider Web.

**Comment by PrabbyDD** [comment]
What is the intent of using the .condition()? Do I need to subtmit some function into it that checks whether the pokemon is a ghost type and then if it queue a message?

**Review by DayKev — DISMISSED** [review]

File: src/test/moves/octolock.test.ts [hunk 1]
```diff
@@ -110,7 +107,25 @@ describe("Moves - Octolock", () => {
     game.move.select(Moves.OCTOLOCK);
 
     // after Octolock - enemy should be trapped
-    await game.phaseInterceptor.to(MoveEndPhase);
+    await game.phaseInterceptor.to("MoveEndPhase");
     expect(enemyPokemon.findTag(t => t instanceof TrappedTag)).toBeDefined();
   });
+
+  it("does not work on ghost type pokemon", async () => {
```
Comment (frutescens) [inline src/test/moves/octolock.test.ts:114]: Could you add an additional test that checks for that this fix applies to a Pokemon affected by Trick O' Treat?

**Review by frutescens — DISMISSED** [review]

File: src/data/move.ts [hunk 2]
```diff
@@ -8717,6 +8721,7 @@ export function initMoves() {
     new SelfStatusMove(Moves.IRON_DEFENSE, Type.STEEL, -1, 15, -1, 0, 3)
       .attr(StatStageChangeAttr, [ Stat.DEF ], 2, true),
     new StatusMove(Moves.BLOCK, Type.NORMAL, -1, 5, -1, 0, 3)
+      .condition(failIfGhostType)
```
Comment (innerthunder) [inline src/data/move.ts:8724]:
```suggestion
      .condition(failIfGhostTypeCondition)
```

File: src/data/move.ts [hunk 3]
```diff
@@ -10012,6 +10017,7 @@ export function initMoves() {
       .attr(EatBerryAttr)
       .target(MoveTarget.ALL),
     new StatusMove(Moves.OCTOLOCK, Type.FIGHTING, 100, 15, -1, 0, 8)
+      .condition(failIfGhostType)
```
Comment (innerthunder) [inline src/data/move.ts:10020]:
```suggestion
      .condition(failIfGhostTypeCondition)
```

**Review by DayKev — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**